pkg/controller/core/queuedworkload_controller.go

```diff
@@ -200,3 +238,66 @@ func workloadFinished(w *kueue.QueuedWorkload) bool {
 	}
 	return false
 }
+
+// GetWorkloadCondition extracts the provided condition from the given status and returns that.
+// Returns nil and -1 if the condition is not present, and the index of the located condition.
+func GetWorkloadCondition(status *kueue.QueuedWorkloadStatus, conditionType kueue.QueuedWorkloadConditionType) (int, *kueue.QueuedWorkloadCondition) {
```

Comment:
The function `workloadFinished` should probably reuse this one.

Also, why is the function exported?

Reply:
> TODO: The function workloadFinished should  reuse `getWorkloadCondition`  

TODO to not diverge from this PR scope?